From f937c4060ed7f0034e2cfadfa7ecc1264e100038 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Tue, 8 Sep 2020 10:01:45 -0700
Subject: [PATCH] Pass --disable-reloc-section on PE targets for PR 25662 test

Pass --disable-reloc-section on PE targets for PR 25662 test since

commit 514b4e191d5f46de8e142fe216e677a35fa9c4bb

Author: Jeremy Drake <sourceware-bugzilla@jdrake.com>
Date:   Thu Aug 27 12:58:27 2020 +0100

    Change the default characteristics of DLLs built by the linker to more secure settings.

defaulted to --enable-reloc-section.

	PR ld/26587
	* testsuite/binutils-all/objcopy.exp: Pass --disable-reloc-section
	to ld on PE targets for PR 25662 test.
---
 binutils/testsuite/binutils-all/objcopy.exp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/binutils/testsuite/binutils-all/objcopy.exp b/binutils/testsuite/binutils-all/objcopy.exp
index dd74860f9e..90ababf905 100644
--- a/binutils/testsuite/binutils-all/objcopy.exp
+++ b/binutils/testsuite/binutils-all/objcopy.exp
@@ -1354,4 +1354,12 @@ if { [istarget pdp11-*-*] } {
     set src "pr25662.s"
 }
 
-objcopy_test "pr25662" $src executable "" "-T$srcdir/$subdir/pr25662.ld"
+set ldflags "-T$srcdir/$subdir/pr25662.ld"
+if { [istarget *-*-cygwin] || [istarget *-*-mingw*] } {
+   append ldflags " --disable-reloc-section"
+}
+
+#xcoff doesn't support arbitrary sections
+if { ![is_xcoff_format] } {
+    objcopy_test "pr25662" $src executable "" $ldflags
+}
-- 
2.28.0.windows.1

